### PR TITLE
Derive serialization for the types in spirv-headers

### PR DESCRIPTION
**Pull request by Nanopro**
Based on #123 

Added serde as a optional dependency in spirv-header,
and feature gated derive for all enums and structs.

### CODE REVIEW

**Comment by kvark** [comment]
@antiagainst could you take a look please?

**Comment by Nanopro** [comment]
@kvark Is it good now?

**Comment by kvark** [comment]
Yes, that's more future-proof, thanks! Please make sure the path is CI tested.
Also, I really want @antiagainst to look at this before merging.

**Comment by Nanopro** [comment]
@kvark I am not sure that error in CI is caused by my commits.

**Comment by kvark** [comment]
indeed! filed #126

File: spirv/Cargo.toml [hunk 1]
```diff
@@ -11,9 +11,14 @@ readme = "README.md"
...
+serialize = ['serde']
+deserialize = ['serde']
+
 [lib]
 path = "lib.rs"
 
 [dependencies]
 bitflags = "1"
 num-traits = "0.2"
+serde = {version = "1", optional = true, features = ["derive"]}
```
Comment (kvark) [inline spirv/Cargo.toml:24]: note that this isn't correct, feature should be `serde_derive`, or the version needs to be more specific

**Comment by kvark** [comment]
@antiagainst how do we get this out for users?

**Comment by antiagainst** [comment]
@kvark: I see we might want more breaking changes along the way so it's better to sort out the versioning issue. I'm thinking maybe it's better to retire the current spirv_header crate and instead publish to spirv crate so that we have a chance to detach the spec version and the crate version. @neon64 asked me previously in #21 but back then we don't have a clear need to make it worthwhile. Now maybe its the time.

**Comment by kvark** [comment]
I haven't given it a lot of thought, tbh. It sounded to me, based on the previous discussions, that relying on upstream Spec semantic versioning can be a problem. In this case, yes, we have to detach it. Do you own "spirv" crate now? Sounds like a good plan :)

**Pull request merged**